src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -35,13 +38,19 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
       .credential(new AzureKeyCredential(getSubscriptionKey))
       .endpoint(getEndpoint)
       .buildClient()
+  var timeout = Duration(10,"sec")
+  var concurrency = 10
 
   protected def transformTextRows(toRow: TAResponseV4[T] => Row)
                                  (rows: Iterator[Row]): Iterator[Row] = {
-    rows.map { row =>
-      val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
-      Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
-    }}
+    val futures = rows.map { row =>
+      Future {
+        lazy val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))

```

Comment:
dont think you need lazy here as the future is call by name which is kindof nifty ;)